core/app/jobs/spree/promotion_code_batch_job.rb

```diff
@@ -6,7 +6,8 @@ class PromotionCodeBatchJob < ActiveJob::Base
 
     def perform(promotion_code_batch)
       PromotionCode::BatchBuilder.new(
-        promotion_code_batch
+        promotion_code_batch,
+        join_characters: promotion_code_batch.join_characters
```

Comment:
since the `join_characters` is included into the `promotion_code_batch` we are initializing the `BatchBuilder` with, what's the purpose of also passing it as a keyword argument? I think it could be better to keep the `.new` identical and work inside the `BatchBuilder` class to make the `join_characters` an instance attribute instead of a class level one